### ISSUE TITLE
Skipping a test at runtime results in duplicate entries in Report Portal

### ISSUE THREAD

**Issue by rdp84** [reportportal/agent-js-mocha#76]
I have a use case in some tests where at runtime we are checking for a condition and if that condition is false we call `this.skip()`. This is described in the Mocha documentation [here](https://mochajs.org/#inclusive-tests); the result of calling `this.skip()` is that Mocha emits the `PENDING` event but since Mocha started the test it has already emitted `TEST_BEGIN` event.

I've noticed though that doing this results in the test being reported twice in Report Portal; one of the entries has the correct status of `Skipped` but the other has a status of `In Progress`.

I had a look into this further and it seems to be with how the `PENDING` event is being handled. I think the function for handling the event begins on this [line](https://github.com/reportportal/agent-js-mocha/blob/45596d32f4b607332fd85a7bda75cb2021f32962/lib/mochaReporter.js#L354) and it first calls `onTestStart` followed by `finishTest`.

For my use case this is a problem. Since I'm skipping at runtime Mocha has started the test so `onTestStart` gets called (creating an entry in Report Portal), the condition fails, `this.skip()` is called emitting the `PENDING` event and we end up in `onTestPending`. `onTestStart` is called, which creates a new entry in Report Portal for the test, followed by `testFinish` which sets the status of the second entry in Report Portal to be `Skipped`. However the first entry still lives on and remains.

The code caters for the use case as marking a test as pending at write time, as described [here](https://mochajs.org/#pending-tests) in Mocha documentation. In this case Mocha does not attempt to start the test so the `TEST_BEGIN` event is nerver emitted and therefore `onTestStart` isn't called.

For my use case I need the `onTestPending` function to check whether it needs to call the `onTestStart` function; I tried with the following update:

```
  onTestPending(test) {
    if (this.currentTest === null) {
      this.onTestStart(test);
    }
    this.finishTest(test, testStatuses.SKIPPED);
  }
```
and that seemed to work but unsure as to whether this would be acceptable.

**Comment by oranje322** [comment]
Hello, @rdp84, 
Thanks for your investigation.

I think your solution is acceptable.

Will be fixed in the next version.

**Comment by AmsterGet** [comment]
Fixed in https://github.com/reportportal/agent-js-mocha/pull/83. Will be released in the v5.0.2.
